api/graphite.go

```diff
@@ -698,6 +701,71 @@ func (s *Server) executePlan(ctx context.Context, orgId uint32, plan expr.Plan)
 	return out, err
 }
 
+// find the best consolidation method based on what was requested and what aggregations are available.
+func closestAggMethod(requested consolidation.Consolidator, available []conf.Method) consolidation.Consolidator {
+	// if there is only 1 consolidation method available, then that is all we can return.
+	if len(available) == 1 {
+		return consolidation.Consolidator(available[0])
+	}
+
+	avail := map[consolidation.Consolidator]struct{}{}
+	for _, a := range available {
+		avail[consolidation.Consolidator(a)] = struct{}{}
+	}
+	var orderOfPreference []consolidation.Consolidator
+	switch requested {
+	case consolidation.Avg:
+		orderOfPreference = []consolidation.Consolidator{
```

Comment:
instead of allocating the `orderOfPreference` and fill it with data for every request we handle, wouldn't it be easier and save some allocations to only once create the list `orderOfPreference` once for each of the configured consolidations in storage-schemas and then use them at each request?

Reply:
good idea.  added a commit to address it.